client/post-editor/post-editor.jsx

```diff
@@ -683,7 +701,7 @@ export class PostEditor extends React.Component {
 
 	iframePreview = async () => {
 		if ( this.props.isDirty ) {
-			this.autosave();
+			await this.autosave();
```

Comment:
I'm assuming you did this because the preview depends on the saved state? It's hard to tell in such a large and complex component.

Reply:
Yes, this is a drive-by fix of a related bug. The `await` ensures that the following scenario works:
1. Edit the post content, don't save it yet.
2. Before any autosave is done, click the "Preview" button.
3. The preview should show the latest content, i.e., the one that's in your editor UI. But to show that, an autosave needs to be done first and we need to wait until it's done before asking the server to show a preview.

Without the `await`, there is a race condition: autosave request is in flight, and in parallel we ask the server to show a preview iframe. The preview will show the old content.

Reply:
Sounds good 👍